server/controllers/gradeController.js

```diff
@@ -105,10 +107,60 @@ async function createRound(req, res, next) {
     }
 }
 
+async function createFixture(req, res, next) {
+    try {
+        const { teams, grade, body: { numRounds, datesAndLocations } } = req
+
+        const numGames = Math.min(Math.floor(teams.length / 2), datesAndLocations.length)
+        const allTeams = teams.map((team) => new TeamNode(team._id))
+
+        for (let i = 0; i < numRounds; i++) {
+            const newRound = await _createRound(grade, next)
+            for (let j = 0; j < numGames; j++) {
+                const team = ejectNextTeamNode(allTeams)
+                const opponent = team.nextTeamToPlay(allTeams)
+
+                team.playTeam(opponent)
+                opponent.playTeam(team)
+
+                const { location, locationName, dateStart, dateFinish } =
+                    ejectNextDateAndLocation(datesAndLocations)
+
+                await _createGame(
+                    team.teamId,
+                    opponent.teamId,
+                    dateStart,
+                    dateFinish,
+                    newRound,
+                    locationName,
+                    location,
+                    next
+                )
+            }
+            const round = await Round.findById(newRound._id)
```

Comment:
For some reason I couldn't use `newRound` that I created above because the games array kept being empty (no idea why). So for now I just re-fetched the round, but I'll try and look into as to why the games array is empty so we can reduce the number of db calls.

Reply:
i think this is because `_createGame` only updates the round doc in the db but that doesn't get reflected in the local var here. if you wanted you could return the new round document from within `_createGame` (i.e. the thing returned by `await Round.findOneAndUpdate` since you have the `{ new: true }` there). would need to update other controllers that use `_createGame` function though but shouldn't be too bad